docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+download queue. Internally, the download queue is used for installing
+models located at remote locations. The queue is implemented by the
+`DownloadQueueService` defined in
+`invokeai.app.services.download_manager`. However, most of the
+implementation is spread out among several files in
+`invokeai/backend/model_manager/download/*`
+
+A default download queue is located in
+`ApiDependencies.invoker.services.download_queue`. However, you can
+create additional instances if you need to isolate your queue from the
+main one.
+
+### A job for every task
+
+The queue operates on a series of download job objects. These objects
+specify the source and destination of the download, and keep track of
+the progress of the download. Jobs come in a variety of shapes and
+colors as they are progressively specialized for particular download
+task.
+
+The basic job is the `DownloadJobBase`, a pydantic object with the
+following fields:
+
+| **Field**      | **Type**        |  **Default**  | **Description** |
+|----------------|-----------------|---------------|-----------------|
+| `id`             | int             |               | Job ID, an integer >= 0           |
+| `priority`       | int             | 10            | Job priority. Lower priorities run before higher priorities |
+| `source`         | Any             |               | Where to download from - untyped in base class |
+| `destination`    | Path            |               | Where to download to              |
+| `status`         | DownloadJobStatus| Idle         | Job's status (see below)          |
+| `event_handlers` | List[DownloadEventHandler]|     | Event handlers (see below)        |
+| `job_started`    | float            |              | Timestamp for when the job started running |
+| `job_ended`      | float            |              | Timestamp for when the job completed or errored out |
+| `job_sequence`   | int              |              | A counter that is incremented each time a model is dequeued |
+| `preserve_partial_downloads`| bool  | False        | Resume partial downloads when relaunched.   |
+| `error`          | Exception        |              | A copy of the Exception that caused an error during download |
+
+When you create a job, you can assign it a `priority`. If multiple
+jobs are queued, the job with the lowest priority runs first. (Don't
+blame me! The Unix developers came up with this convention.) 
+
+Every job has a `source` and a `destination`. `source` is an untyped
+object in the base class, but subclassses redefine it more
+specifically.
+
+The `destination` must be the Path to a file or directory on the local
+filesystem. If the Path points to a new or existing file, then the
+source will be stored under that filename. If the Path ponts to an
+existing directory, then the downloaded file will be stored inside the
+directory, usually using the name assigned to it at the remote site in
+the `content-disposition` http field.
```

Comment:
I agree and will fix this.

However, to support resumption of partial downloads we do want to detect that there is a partial file of the same name present in the directory. So I can add a flag to the argument list indicating that the user wants to resume a partial download rather than raise an exception on a name collision. This would provide for future compatibility with a robust resume mechanism. What do you think?